src/multiplex.c

```diff
@@ -35,33 +35,49 @@
 
 #define BUF_SIZE 4096
 
-int child_status = -1;
+static volatile int child_status = -1;
+static bool is_client;
 
 struct thread_args{
     int multi_fd;
     int *fds;
     int fds_count;
 };
 
-void sigchld_handler(int arg __attribute__((__unused__)))
+static void sigchld_handler(int arg __attribute__((__unused__)))
 {
     int stat_loc;
     wait(&stat_loc);
     if (WIFEXITED(stat_loc))
         child_status = WEXITSTATUS(stat_loc);
+    else if (WIFSIGNALED(stat_loc))
+        child_status = 128 + WTERMSIG(stat_loc);
 }
 
-void setup_sigchld(void)
+static void sigpipe_handler(int arg __attribute__((__unused__))) {}
+
+void setup_sigchld(bool arg)
 {
     struct sigaction sa;
+    memset(&sa, 0, sizeof sa);
 
+    is_client = arg;
     sa.sa_handler = sigchld_handler;
     sa.sa_flags = 0;
     sigemptyset(&sa.sa_mask);
-    sigaction(SIGCHLD, &sa, NULL);
+    if (sigaction(SIGCHLD, &sa, NULL) != 0) {
+        perror("sigaction");
+        exit(1);
+    }
+
+    sa.sa_handler = sigpipe_handler;
```

Comment:
```suggestion
    sa.sa_handler = SIG_IGN;
```
? but then, you'd need to reset it before exec...